CMakeLists.txt

```diff
@@ -84,9 +84,9 @@ if(ENABLE_CCACHE)
 	find_program(CCACHE ccache REQUIRED)
 endif()
 
-# On Linux, use ccache via CMAKE_CXX_COMPILER_LAUNCHER.
+# On Linux and MinGW, use ccache via CMAKE_CXX_COMPILER_LAUNCHER.
 # The XCode and MSVC builds each require some more configuration further down.
-if(ENABLE_CCACHE AND LINUX)
+if(ENABLE_CCACHE AND (LINUX OR MINGW))
```

Comment:
iirc there's a `GNU` check

Reply:
I think that'd include installations of GCC on osx, which is not something I've tested

Reply:
which is not something anyone has ever tested :)

Reply:
yep, so let's keep `MinGW`

Reply:
well, what about Makefile or Ninja generator on macOS regardless of the compiler? They should also work by simply specifying the "compiler launcher".

I'd rather just add a check for those generators here (maybe excluding Windows OS, not sure how ccache works there in command-line builds).

Reply:
Not sure what you mean by
>maybe excluding Windows OS

this PR _is_ for Windows, using the `gcc` port [mingw-w64](https://www.mingw-w64.org/) that is used on host-os Windows targeting Windows (not cross-compilation). I know mingw can be used to cross-compile on Linux targeting Windows, but that's unrelated to this PR. I could rewrite it into
```
if(ENABLE_CCACHE AND (LINUX OR (WINDOWS AND MINGW)))
```

so I don't wind up unintentionally supporting ccache for cross-compilation.. but I thought that was a nice side-effect so I just skipped checking for `WINDOWS`. 

> well, what about Makefile or Ninja generator on macOS regardless of the compiler?
It doesn't matter if you use Makefile or Ninja generator. This PR is enabling ccache for `windows-mingw-release`, both Makefile and Ninja generator. 

I just now found out there's no `GNU`-variable in CMake (I also thought it did exist), at least not listed here: https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html

I'm all for supporting other platforms with ccache :), but it's out of scope of this PR

Reply:
my main remark is that instead of adding every single platform or compiler here we should instead check the CMake generator that's used. I definitely know that ccache works through the standard compiler launcher variables for Makefile and Ninja generators, and definitely neither for Xcode nor for Visual Studio (that's already handled separately).
 
> this PR _is_ for Windows

right, was thinking about msvc :)

Reply:
> my main remark is that instead of adding every single platform or compiler here we should instead check the CMake generator that's used.

Ah, now I see what you mean. I'll make a new commit.